crates/oxc_linter/src/rules/eslint/no_useless_call.rs

```diff
@@ -0,0 +1,156 @@
+use oxc_ast::{ast::{Argument, CallExpression, ChainElement, Expression, StaticMemberExpression}, match_member_expression, visit::walk::walk_expression, AstKind};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{ast_util::*, context::LintContext, rule::Rule, rules::jest::expect_expect, utils::is_same_reference, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUselessCall;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// 
+    /// Disallow unnecessary calls to `.call()` and `.apply()`
+    ///
+    /// ### Why is this bad?
+    /// 
+    /// This rule is aimed to flag usage of Function.prototype.call() and Function.prototype.apply() that can be replaced with the normal function invocation.
```

Comment:
Can you elaborate more here? Maybe take the explanation from ESLint's website?

Reply:
The comment is from [eslint website](https://eslint.org/docs/latest/rules/no-useless-call).
I think it will [do more think](https://262.ecma-international.org/5.1/#sec-15.3.4.4) than normal function.